device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/FileUpload.java

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.azure.sdk.iot.device;
+
+import com.microsoft.azure.sdk.iot.deps.serializer.FileUploadCompletionNotification;
+import com.microsoft.azure.sdk.iot.deps.serializer.FileUploadSasUriRequest;
+import com.microsoft.azure.sdk.iot.deps.serializer.FileUploadSasUriResponse;
+import com.microsoft.azure.sdk.iot.device.IotHubMethod;
+import com.microsoft.azure.sdk.iot.device.IotHubStatusCode;
+import com.microsoft.azure.sdk.iot.device.ResponseMessage;
+import com.microsoft.azure.sdk.iot.device.exceptions.IotHubServiceException;
+import com.microsoft.azure.sdk.iot.device.transport.IotHubTransportMessage;
+import com.microsoft.azure.sdk.iot.device.transport.https.HttpsTransportManager;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+
+@Slf4j
+public final class FileUpload
+{
+    private static final Charset DEFAULT_IOTHUB_MESSAGE_CHARSET = StandardCharsets.UTF_8;
+
+    private final HttpsTransportManager httpsTransportManager;
+
+    FileUpload(HttpsTransportManager httpsTransportManager)
+    {
+        this.httpsTransportManager = httpsTransportManager;
+        httpsTransportManager.open();
+    }
+
+    FileUploadSasUriResponse getFileUploadSasUri(FileUploadSasUriRequest request) throws IOException
+    {
+        IotHubTransportMessage message = new IotHubTransportMessage(request.toJson());
+        message.setIotHubMethod(IotHubMethod.POST);
+
+        ResponseMessage responseMessage = httpsTransportManager.getFileUploadSasUri(message);
+
+        String responseMessagePayload = validateServiceStatusCode(responseMessage, "Failed to get the file upload SAS URI");
+
+        if (responseMessagePayload == null || responseMessagePayload.isEmpty())
+        {
+            throw new IOException("Sas URI response message had no payload");
+        }
+
+        return new FileUploadSasUriResponse(responseMessagePayload);
+    }
+
+    @SuppressWarnings("UnusedReturnValue") // Public method
+    IotHubStatusCode sendNotification(FileUploadCompletionNotification fileUploadStatusParser) throws IOException
+    {
+        IotHubTransportMessage message = new IotHubTransportMessage(fileUploadStatusParser.toJson());
+        message.setIotHubMethod(IotHubMethod.POST);
+
+        ResponseMessage responseMessage = httpsTransportManager.sendFileUploadNotification(message);
+        httpsTransportManager.close();
```

Comment:
unrelated to this PR, but any reason why we close the http layer after sending the upload notification? 
what if a customer want to perform > retrieve uri for file 1 -> upload file 1 -> notify completion for file 1 -> retrieve uri for file 2 -> ..

Reply:
Both open and close are basically no-ops since the HttpsTransportManager doesn't persist anything between requests. That said, there is no reason to close here. I'll just open the HttpsTransportManager when FileUpload is constructed, and close it when FileUpload is closed.